python/ray/data/_internal/progress_bar.py

```diff
@@ -167,9 +170,13 @@ def fetch_until_complete(self, refs: List[ObjectRef]) -> List[Any]:
             total_rows_processed = 0
             for ref, result in zip(done, ray.get(done)):
                 ref_to_result[ref] = result
-                num_rows = (
-                    result.num_rows if hasattr(result, "num_rows") else 1
-                )  # Default to 1 if no row count is available
+                if hasattr(result, "num_rows"):
+                    num_rows = result.num_rows
+                elif hasattr(result, "__len__"):
+                    # For output is DataFrame,i.e. sort_sample
+                    num_rows = len(result)
+                else:
+                    num_rows = 1
```

Comment:
could you consolidate this logic and the one in `block_until_complete()` as a static method in `ProgressBar` or a utility method in the file?